### PR TITLE
[ISSUE #199] Kafka Connector: Initial Implementation of Consumer & Producer

### PR DESCRIPTION
**Pull request by Markliniubility**
Impl ISSUE #199

### Motivation

*The goal is to expand the event storage component that EventMesh can support, Kafka, based on the existing API. This PR is the first step of this project, which implemented a consumer.*

*This PR, as well as all the following PR on Kafka Connectors, will not be merged into the master. It will be merged into a Kafka branch instead*

### Modifications

*A basic/mimimal consumer of Kafka.*

### Documentation

- Does this pull request introduces a new feature? (yes / no)
- Yes
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- Not yet documented
- If a feature is not applicable for documentation, explain why?
- It will be documented after the completion of the producer and a functional Kafka connector
- If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation
- The follow-up will be creating the producer, as well as implementing more features of Kafka

### CODE REVIEW

File: eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/producer/ProducerImpl.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+package org.apache.eventmesh.connector.kafka.producer;
+
+//import org.apache.eventmesh.api.RequestReplyCallback;
+//import org.apache.eventmesh.api.SendCallback;
+//import org.apache.eventmesh.api.SendResult;
+//import org.apache.eventmesh.api.exception.ConnectorRuntimeException;
+//import org.apache.eventmesh.api.exception.OnExceptionContext;
+//import org.apache.eventmesh.api.producer.Producer;
+//import org.apache.eventmesh.common.Constants;
```
Comment (Alonexc) [inline eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/producer/ProducerImpl.java:26]: Please remove the invalid code.

**Comment by Alonexc** [comment]
The license check did not pass, please check the license format or if you forgot to add a license.

**Comment by mytang0** [comment]
Hello, please fix the license check problem.

**Comment by LIU-WEI-git** [comment]
This pr just implements consumer without enough tests. I recommend you implement producer first and verify it through local kafka server. If you could write some UT test with kafka testcontainer, it will be much better. After that, we could have a base to implement consumer correctly.

What's more, please replace rocketmq with kafka in your code and comment.

**Comment by Markliniubility** [comment]
> Hi, thanks for your initial implementation. Here are some suggestions.
> 
> This pr just implements consumer without enough tests. I recommend you implement producer first and verify it through local kafka server. If you could write some UT with kafka testcontainer, it will be much better. After that, we could have a base to implement consumer correctly.
> 
> What's more, please replace rocketmq with kafka in your code and comment.

Thank you for reviewing! I agree with you on implementing the producer first (which I should have done first) and providing unit tests, and I am working on implementing the producer. However, the branch was only going to merge into a development branch, kafka-connector, instead of the master branch. I was going to add the unit tests and producer in separate PRs and make sure everything works and is fully tested before rolling out to the master branch. I prefer splitting the project into [small PR](https://google.github.io/eng-practices/review/developer/small-cls.html)s since it can be more easily reviewed and understood.

**Pull request merged**